evm/src/executor/opts.rs

```diff
@@ -40,15 +40,46 @@ impl EvmOpts {
     pub fn evm_env(&self) -> revm::Env {
         if let Some(ref fork_url) = self.fork_url {
             let rt = RuntimeOrHandle::new();
-            let provider =
-                Provider::try_from(fork_url.as_str()).expect("could not instantiated provider");
-            let fut =
-                environment(&provider, self.env.chain_id, self.fork_block_number, self.sender);
-            match rt {
-                RuntimeOrHandle::Runtime(runtime) => runtime.block_on(fut),
-                RuntimeOrHandle::Handle(handle) => handle.block_on(fut),
+
+            if fork_url.starts_with("http") {
+                let fut = async {
+                    let provider = Provider::try_from(fork_url.as_str())
+                        .expect("could not instantiated provider");
+                    environment(&provider, self.env.chain_id, self.fork_block_number, self.sender)
+                        .await
+                };
+                match rt {
+                    RuntimeOrHandle::Runtime(runtime) => runtime.block_on(fut),
+                    RuntimeOrHandle::Handle(handle) => handle.block_on(fut),
+                }
+                .expect("could not instantiate forked environment")
```

Comment:
this getting a bit messy.
we have already a `block_on` in `cli/utils` I believe.

let's either replicate that for this crate or better make this function async so we only have to do it once for the whole `evm_env` function.

Reply:
imho this function should be async & there is too much `rt.block_on` in  `Fork::spawn_backend` too.